### PR TITLE
Project setup, database setup, migration configuration

### PR DESCRIPTION
**Pull request by mmishv**

### CODE REVIEW

File: src/database.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        )
+        async with async_session_maker() as session:
+            try:
+                yield session
+            finally:
+                await session.close()
+
+
+@asynccontextmanager
+async def get_db():
```
Comment (lihoman) [inline src/database.py:36]: where do you use this connection?

File: src/models.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    USER = "USER"
+    ADMIN = "ADMIN"
+    MODERATOR = "MODERATOR"
+
+
+class Group(Base):
+    __tablename__ = "group"
+
+    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
+    name: Mapped[str] = mapped_column(String(100), index=True)
```
Comment (lihoman) [inline src/models.py:23]: use Text instead of string with length constrains

File: src/models.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
+
+    users = relationship(
+        "User", back_populates="group", order_by="User.username", lazy="joined"
+    )
+
+
+class User(Base):
```
Comment (lihoman) [inline src/models.py:31]: replace String(...) on Text

File: src/models.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    name: Mapped[str]
+    surname: Mapped[str]
+    username: Mapped[str] = mapped_column(String(100), unique=True, index=True)
+    phone_number: Mapped[str] = mapped_column(String(20), unique=True)
+    email: Mapped[str] = mapped_column(String(256), unique=True, index=True)
+    role: Mapped[RoleEnum] = mapped_column(Enum(RoleEnum), default=RoleEnum.USER)
+    group_id: Mapped[int] = mapped_column(Integer, ForeignKey("group.id"))
+    image_s3_path: Mapped[str] = mapped_column(String(1024))
+    is_blocked: Mapped[bool] = mapped_column(default=False)
+    created_at: Mapped[datetime]
```
Comment (lihoman) [inline src/models.py:46]: default value

File: src/models.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    modified_at: Mapped[datetime]
```
Comment (lihoman) [inline src/models.py:47]: default value on update

File: src/other/router.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    return {"result": "You've successfully checked your health!"}
+
+
+@router.get("/groups/")
+async def read_groups(db_session: AsyncSession = Depends(create_async_session)):
+    logger.info("Read all groups endpoint accessed")
+    async with db_session as conn:
+        query = select(Group).options(selectinload(Group.users))
+        result = await conn.execute(query)
+    return result.scalars().all()
```
Comment (lihoman) [inline src/other/router.py:28]: add validation (pydantic schema)

File: src/settings.py
Comment (lihoman) [inline src/settings.py:1]: read about python-dotenv

File: src/other/router.py
Comment (lihoman) [inline src/other/router.py:1]: store business-logic in separate module 
in routers you only have to call a separate function

File: src/other/router.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+
+@router.get("/groups/{group_id}")
+async def read_group(
+    group_id: int, db_session: AsyncSession = Depends(create_async_session)
+):
+    logger.info(f"Read group with id = {group_id} endpoint accessed")
+    async with db_session as conn:
+        group = await conn.execute(select(Group).filter(Group.id == group_id))
```
Comment (lihoman) [inline src/other/router.py:37]: split in query = ... and after that conn.execute(query) like you have in read_groups endpoint

File: src/other/router.py
Comment (lihoman) [inline src/other/router.py:1]: do we need these endpoints? you have a list in task's requirements with final APIs, check please that table again

**Comment by mmishv** [comment]
This pull request is a work in progress.

**Pull request closed without merging**